client/src/components/Calendar.js

```diff
@@ -1,78 +1,78 @@
+        listDay: "list day"
+      }}
+      defaultView="dayGridMonth"
+      views={{
+        timeGridWeek: {
+          eventLimitClick: "day"
+        },
+        dayGrid: {
+          eventLimitClick: "popover"
+        }
+      }}
+      allDayDefault={false}
+      eventTimeFormat={{
+        hour: "2-digit",
+        minute: "2-digit",
+        meridiem: false,
+        omitZeroMinute: false,
+        hour12: false
+      }}
+      slotLabelFormat={{
+        hour: "2-digit",
+        minute: "2-digit",
+        meridiem: false,
+        omitZeroMinute: false,
+        hour12: false
+      }}
+      height="auto" // assume a natural height, no scrollbars will be used
+      aspectRatio={3} // ratio of width-to-height
+      timeGridEventMinHeight={20}
+      ref={calendarComponentRef}
+      events={events}
+      eventOverlap
+      eventLimit
+      eventClick={eventInfo => {}}
+      dateClick={info => {
+        const calendarApi = calendarComponentRef.current.getApi()
+        calendarApi.changeView("listDay", info.dateStr) // call a method on the Calendar object
+      }}
+    />
+  )
+}
 
-  static defaultProps = {
-    events: []
-  }
+Calendar.propTypes = {
+  events: PropTypes.func.isRequired,
+  calendarComponentRef: PropTypes.shape({
+    current: PropTypes.oneOfType([PropTypes.func, PropTypes.object])
+  })
+}
 
-  render() {
-    const { events, calendarComponentRef } = this.props
-    return (
-      <FullCalendar
-        plugins={[dayGridPlugin, timeGridPlugin, listPlugin, interactionPlugin]}
-        header={{
-          left: "prev,next today filterDraft",
-          center: "title",
-          right: "dayGridMonth,timeGridWeek,timeGridDay,listDay"
-        }}
-        buttonText={{
-          listDay: "list day"
-        }}
-        defaultView="dayGridMonth"
-        views={{
-          timeGridWeek: {
-            eventLimitClick: "day"
-          },
-          dayGrid: {
-            eventLimitClick: "popover"
-          }
-        }}
-        allDayDefault={false}
-        eventTimeFormat={{
-          hour: "2-digit",
-          minute: "2-digit",
-          meridiem: false,
-          omitZeroMinute: false,
-          hour12: false
-        }}
-        slotLabelFormat={{
-          hour: "2-digit",
-          minute: "2-digit",
-          meridiem: false,
-          omitZeroMinute: false,
-          hour12: false
-        }}
-        height="auto" // assume a natural height, no scrollbars will be used
-        aspectRatio={3} // ratio of width-to-height
-        timeGridEventMinHeight={20}
-        ref={calendarComponentRef}
-        events={events}
-        eventOverlap
-        eventLimit
-        eventClick={eventInfo => {}}
-        dateClick={info => {
-          let calendarApi = calendarComponentRef.current.getApi()
-          calendarApi.changeView("listDay", info.dateStr) // call a method on the Calendar object
-        }}
-      />
-    )
-  }
+Calendar.defaultProps = {
+  events: []
```

Comment:
This can be removed now right? Events are a required function now.

Reply:
Yup, good point.